### PR TITLE
When opening new file, makes sure it is created

### LINKED ISSUE #725: New file behavior is a bit confusing
When I click on a markdown link for a non-existent file and the "automatically create files for link targets" option is checked off, Mac Down will open a new, blank editor. Perfect. However, upon entering text into the blank editor, I receive a message that says this:

> The file <filename> cannot be found.
> You can duplicate this document and edit the duplicate. Only the duplicate will include your changes.

Also the editor is locked. I don't think this should be the behavior, but maybe I'm not understanding the use case you're supporting. I would like to begin editing, and then once the buffer is full of things that I want to save to that file, I should be able to File->Save and then the file will be created.

There is a work around. If I immediately File->Save with the buffer empty, it will create the file and I will be able to continue editing. However this work around, of saving a blank file, is pretty counter to any program I've used that's capable of doing a file safe. If this is not a necessary behavior I think it would be great if it were changed to align itself with the user experience of more common editors.

### PR DESCRIPTION
**Pull request by brunophilipe**
Renames `openUntitledDocumentForURL:display:error:` to `createNewEmptyDocumentForURL:display:error:` and makes it create the file specified in the `url` parameter.

Also adds a new test when creating a file when a markdown file link is clicked: The current file where the link was clicked must exist in the file system.

Fixes #725.

### CODE REVIEW

File: MacDown/Code/Document/MPDocument.m [hunk 1]
```diff
@@ -1644,29 +1644,51 @@ - (void)openOrCreateFileForUrl:(NSURL *)url
...
-        return;
     }
-
-    // Try to open it.
-    if ([[NSWorkspace sharedWorkspace] openURL:url])
-        return;
+    else
+    {
+        // Try to open it.
+        if ([[NSWorkspace sharedWorkspace] openURL:url])
```
Comment (franklinyu) [inline MacDown/Code/Document/MPDocument.m]: It can be an `else if` here?

**Comment by brunophilipe** [comment]
Actually, that `if` statement is pointless. Fixed now :)

**Comment by brunophilipe** [comment]
Makes sense! I've made the requested changes.

**Comment by uranusjr** [comment]
Merged via cd4228c. I refactored `-openOrCreateFileForUrl:` a little to flatten the structure a bit (a95b85c).

**Pull request closed without merging**